pkg/cli/completion.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+// TODO: support fish code completion
+//  At the time this comment is written, the imported spf13.cobra version does not support fish completion.
+//  However, fish completion has been added to new spf13.cobra versions. When a new spf13.cobra version that
+//  supports it is used, the changes indicated with `NOTE(fish)` should be applied to support it.
+
+func (c *cli) newCompletionCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:                   "completion [bash|zsh|powershell]", // NOTE(fish): add fish
+		DisableFlagsInUseLine: true,
+		ValidArgs:             []string{"bash", "zsh", "powershell"}, // NOTE(fish): add fish
+		Args:                  cobra.ExactValidArgs(1),
+		Short:                 "Generate completion script",
+		Long: fmt.Sprintf(`To load completions:
+
+Bash:
+
+$ source <(%[1]s completion bash)
+
+# To load completions for each session, execute once:
+Linux:
+  $ %[1]s completion bash > /etc/bash_completion.d/%[1]s
+MacOS:
+  $ %[1]s completion bash > /usr/local/etc/bash_completion.d/%[1]s
+
+Zsh:
+
+# If shell completion is not already enabled in your environment you will need
+# to enable it.  You can execute the following once:
+
+$ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+# To load completions for each session, execute once:
+$ %[1]s completion zsh > "${fpath[1]}/_%[1]s"
+
+# You will need to start a new shell for this setup to take effect.
+`, c.commandName),
+		/* NOTE(fish): add the following description to Long
+		   `
+		   Fish:
+
+		   $ %[1]s completion fish | source
+
+		   # To load completions for each session, execute once:
+		   $ %[1]s completion fish > ~/.config/fish/completions/%[1]s.fish
+		   `
+		*/
```

Comment:
IMO we should remove this comments and instead of that add : 

// todo: when the cobra version be upgraded to X add the fish option as described in : URL